storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -89,14 +90,16 @@ public class TopicBasedRemoteLogMetadataManager implements RemoteLogMetadataMana
     private volatile RemoteLogMetadataTopicPartitioner rlmTopicPartitioner;
     private final Set<TopicIdPartition> pendingAssignPartitions = Collections.synchronizedSet(new HashSet<>());
     private volatile boolean initializationFailed;
+    private final Supplier<RemotePartitionMetadataStore> remoteLogMetadataManagerSupplier;
 
     public TopicBasedRemoteLogMetadataManager() {
```

Comment:
Could you add comments to say this default constructor is required as we create `RemoteLogMetadataManager` dynamically?

Reply:
Sure, addressed both in [a8ba568](https://github.com/apache/kafka/pull/15885/commits/a8ba568bb444d0f0b852c6baf63eb343bd8eee71)